environment_setup/tools/tbfc.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+# Define TBOTS_DIR to be the root of the Software repo in your .bashrc/.bash_profile
+ProgName=tb
+  
+sub_help(){
+    echo "Usage: $ProgName <subcommand> [options]\n"
+    echo "Subcommands:"
+    echo "    build {ai, fw} [opts]"
+    echo "    run  {ai, fw} [opts] "
+    echo "    macos {pfw, ufw, ai} [opts] ### EXPERIMENTAL ###"
+    echo ""
+    echo "For help with each subcommand run:"
+    echo "$ProgName <subcommand> -h|--help"
+    echo ""
+}
+  
+sub_build(){
+    echo "Running 'build' command."
+    sub=$1
+    shift
+    echo $@
+    case $sub in
+        "fw")
+            bazel build //firmware_new/... --cpu=stm32h7;
+            ;;
+        "ai")
+            bazel build //software/...;
+            ;;
+        *)
+    esac
+
+}
+  
+sub_run(){
+    sub=$1
+    shift
+    echo $@
+    case $sub in
+        "fw")
+            bazel run --cpu=stm32h7 --compilation_mode=dbg //firmware_new/tools:debug_firmware_on_arm_board $@;
+            ;;
+        "ai")
+            bazel run //software/full_system $@;
+            ;;
+        *)
+    esac
+}
+
+sub_macos(){
+    sub=$1
+    shift
+    echo $@
+    case $sub in
+        "pfw")
+            # Patch workspace 
+            git apply $TBOTS_DIR/environment_setup/tools/macos_fw.patch
+            ;;
+        "ufw")
+            # Unpatch 
+            git apply -R $TBOTS_DIR/environment_setup/tools/macos_fw.patch
+            ;;
+        "ai")
+            echo "Sorry, no support for ai yet :("
+            exit 1
+            ;;
+        *)
+    esac
+}
+  
+function tb() {
+    subcommand=$1
+    case $subcommand in
+        "-h" | "--help")
+            sub_help
+            ;;
+        "")
+            cd $TBOTS_DIR;
+            ;;
+        *)
+            shift
+            cd $TBOTS_DIR/src;
+            sub_${subcommand} $@
+            if [ $? = 127 ]; then
+                echo "Error: '$subcommand' is not a known subcommand." >&2
+                echo "       Run '$ProgName --help' for a list of known subcommands." >&2
+                exit 1
+            fi
+            ;;
+    esac
+}
```

Comment:
I'm not a huge fan of this script, just because I would much prefer that people have to use proper bazel commands so they understand what's going on.

Reply:
This definitely is far from a mandatory script, only for those who are interested; was inspired by the intricate wrapper scripts some of my coworkers have come up with. The main purpose was originally as a way for macOS firmware development, but I realized it could be more useful as a general-purpose script.
@MathewMacDougall  and I discussed having an "unofficial/experimental" folder in `environment_setup` for stuff like this

Reply:
I still think this will obfuscate things more then it helps. To be frank, I want to people to be forced to use the bazel commands so they understand what's happening. I am afraid that if people use this, they won't develop the required understanding to work flexibly with bazel, which is an essential skill for working in our codebase now.